testplan/web_ui/testing/src/Report/BatchReportBeta/state/uiSlice.js

```diff
@@ -0,0 +1,93 @@
+import _isObject from 'lodash/isObject';
+import { createSlice } from '@reduxjs/toolkit/dist/redux-toolkit.esm';
+import * as filterStates from '../../../Common/filterStates';
+import { fetchReport } from './reportActions';
+import setSelectedEntry from './uiActions/setSelectedEntry';
+import { COLUMN_WIDTH } from '../../../Common/defaults';
+
+const __DEV__ = process.env.NODE_ENV !== 'production';
+const FILTER_STATES_ARR = Object.values(filterStates);
+
+/** This state slice contains information specific to how the UI should look */
+export default createSlice({
+  name: 'ui',
+  initialState: {
+    hashAliasToComponent: {},
+    hashComponentToAlias: {},
+    isShowHelpModal: false,
+    isDisplayEmpty: true,
+    filter: filterStates.ALL,
+    isShowTags: false,
+    isShowInfoModal: false,
+    selectedEntry: null,
+    isSettingSelectedEntry: false,
+    doAutoSelect: true,
+    // these are strings for now
+    // TODO: make this sane by storing only one unit type as int
+    sidebarWidthPx: null,
+    sidebarWidthEm: `${COLUMN_WIDTH}em`,
+  },
+  reducers: {
+    setSidebarWidth: {
+      reducer(state, { payload }) { state.sidebarWidthPx = payload; },
+      prepare: (widthPxStr) => ({ payload: widthPxStr }),
+    },
+    setHashComponentAlias: {
+      reducer(state, { payload: { component, alias } }) {
+        state.hashAliasToComponent[alias] = component;
+        state.hashComponentToAlias[component] = alias;
+      },
+      prepare: (component, alias) => ({ payload: { component, alias } }),
+    },
+    setShowTags: {
+      reducer(state, { payload }) { state.isShowTags = payload; },
+      prepare: (showTags = false) => ({ payload: !!showTags }),
+    },
+    setShowInfoModal: {
+      reducer(state, { payload }) { state.isShowInfoModal = payload; },
+      prepare: (showInfoModal = false) => ({ payload: !!showInfoModal }),
+    },
+    setDoAutoSelect: {
+      reducer(state, { payload }) { state.doAutoSelect = payload; },
+      prepare: (doAutoSelect = true) => ({ payload: !!doAutoSelect }),
+    },
+    setFilter: {
+      reducer(state, { payload }) { state.filter = payload; },
+      prepare: (filter = filterStates.ALL) => ({
+        payload: FILTER_STATES_ARR.includes(filter) ? filter : filterStates.ALL
+      }),
+    },
+    setDisplayEmpty: {
+      reducer(state, { payload }) { state.isDisplayEmpty = payload; },
+      prepare: (displayEmpty = true) => ({ payload: !!displayEmpty }),
+    },
+    setShowHelpModal: {
+      reducer(state, { payload }) { state.isShowHelpModal = payload; },
+      prepare: (showHelpModal = false) => ({ payload: !!showHelpModal }),
+    },
+  },
+  extraReducers: {
+    [fetchReport.fulfilled.type](state, action) {
+      state.selectedEntry = action.payload;
+    },
```

Comment:
The same action is handled in reportSlice as well: state.document = action.payload
So are we setting selectedEntry to report root here?